tests/test_bundle.py

```diff
@@ -0,0 +1,92 @@
+import shlex
+import time
+
+import pytest
+from helpers import get_ingress_ip_1dot4, get_ingress_url
+from lightkube.models.rbac_v1 import PolicyRule
+from lightkube.resources.core_v1 import Service
+from lightkube.resources.rbac_authorization_v1 import Role
+from pytest_operator.plugin import OpsTest
+
+USERNAME = "admin"
+PASSWORD = "foobar"
+
+
+@pytest.mark.v1dot6
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_deploy_1dot6(ops_test, lightkube_client, deploy_cmd):
+    print(f"Deploying bundle to {ops_test.model_full_name} using cmd '{deploy_cmd}'")
+    rc, stdout, stderr = await ops_test.run(*shlex.split(deploy_cmd))
+
+    print("Waiting for bundle to be ready")
+    await ops_test.model.wait_for_idle(
+        status="active",
+        raise_on_blocked=False,
+        raise_on_error=True,
+        timeout=3000,
+    )
+    url = get_ingress_url(lightkube_client, ops_test.model_name)
+
+    print("Update Dex and OIDC configs")
+    await ops_test.model.applications["dex-auth"].set_config(
+        {"public-url": url, "static-username": USERNAME, "static-password": PASSWORD}
+    )
+    await ops_test.model.applications["oidc-gatekeeper"].set_config({"public-url": url})
+
+    await ops_test.model.wait_for_idle(
+        status="active",
+        raise_on_blocked=False,
+        raise_on_error=True,
+        timeout=600,
+    )
```

Comment:
This code is shared between 1dot4 and 1dot6 (and likely future versions, too), can we wrap it in a helper?  Maybe like:

```python
def update_dex_oidc_and_wait(url, username, password):
      print("Update Dex and OIDC configs")
    await ops_test.model.applications["dex-auth"].set_config(
        {"public-url": url, "static-username": USERNAME, "static-password": PASSWORD}
    )
    await ops_test.model.applications["oidc-gatekeeper"].set_config({"public-url": url})

    await ops_test.model.wait_for_idle(
        status="active",
        raise_on_blocked=False,
        raise_on_error=True,
        timeout=600,
    )
```

Reply:
We are already creating one deploy function per release version. I didn't want them to share a helper function. In general, I'd like to be able to read a test from top to bottom. It's only used twice right now. It would be more compelling to factor them out when we have one more occurrence.

Reply:
Even reusing it only twice, this seems to me like a good place for a helper.  I don't see the benefit in delaying that until we release 1.7.  

I also don't think helpers need to be confined to just reuse.  I understand a helper hides some of the implementation details and makes it harder to read the entire detail from top to bottom, but in many cases that's actually a benefit.  You are very familiar with the code right now because you're writing it, but others reading this will not be.  When as a reader I see setting some configs and waiting, I don't know why unless I already know how to use dex-auth and oidc-gatekeeper.  Helpers make that read-through experience simpler.  It is harder for a reader to get the full implementation details during a read through, but it is easier to get the *purpose and intent* without having to infer it themselves.  

Taking the opposite view and ignoring how `.wait_for_idle()` is useful across many tests, just using the `.wait_for_idle()` instead of [including all the code](https://github.com/juju/python-libjuju/blob/fd94f0635a36518573a209a93f5361d3152c6df4/juju/model.py#L2580) makes the top-to-bottom read of this test less detailed but I think much easier.  Although not as pronounced, same is true here for setting dex and oidc configs imo

Reply:
Even reusing it only twice, this seems to me like a good place for a helper.  I don't see the benefit in delaying that until we release 1.7.  

I also don't think helpers need to be confined to just reuse.  I understand a helper hides some of the implementation details and makes it harder to read the entire detail from top to bottom, but in many cases that's actually a benefit.  You are very familiar with the code right now because you're writing it, but others reading this will not be.  When as a reader I see setting some configs and waiting, I don't know why unless I already know how to use dex-auth and oidc-gatekeeper.  Helpers make that read-through experience simpler.  It is harder for a reader to get the full implementation details during a read through, but it is easier to get the *purpose and intent* without having to infer it themselves.  

Taking the opposite view and ignoring how `.wait_for_idle()` is useful across many tests, just using the `.wait_for_idle()` instead of [including all the code](https://github.com/juju/python-libjuju/blob/fd94f0635a36518573a209a93f5361d3152c6df4/juju/model.py#L2580) makes the top-to-bottom read of this test less detailed but I think much easier.  Although not as pronounced, same is true here for setting dex and oidc configs imo